dotlottie-rs/src/state_machine/events/mod.rs

```diff
@@ -12,7 +12,21 @@ pub enum Event {
     OnPointerUp { x: f32, y: f32 },
     OnPointerMove { x: f32, y: f32 },
     OnPointerEnter { x: f32, y: f32 },
-    OnPointerExit,
+    OnPointerExit { x: f32, y: f32 },
+    OnComplete,
+    SetNumericContext { key: String, value: f32 },
+}
+
+#[derive(Debug, Clone)]
+pub enum InternalEvent {
+    Bool { value: bool },
+    String { value: String },
+    Numeric { value: f32 },
+    OnPointerDown { target: Option<String> },
+    OnPointerUp { target: Option<String> },
```

Comment:
[question]
The internal and external pointer event are very confusing, specially in the way they are defined. 
for example: OnPointerUp internally has a target of Option<String> but in the external takes x, y as f32. 
Maybe can you help me understand the reason for choosing this approach ?